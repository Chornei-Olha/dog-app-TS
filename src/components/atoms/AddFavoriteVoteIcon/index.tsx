import AddFavoriteVoteDefaultIcon from '../../../assets/icons/AddFavoriteVoteDefaultIcon.svg';
import AddFavoriteVoteHoverIcon from '../../../assets/icons/AddFavoriteVoteHoverIcon.svg';
import AddFavoriteVoteActiveIcon from '../../../assets/icons/AddFavoriteVoteActiveIcon.svg';

type AddFavoriteVoteIconProps = {
  state?: 'hover' | 'active' | 'default';
};

export const AddFavoriteVoteIcon = ({
  state = 'default'
}: AddFavoriteVoteIconProps) => {
  switch (state) {
    case 'hover': {
      return <img src={AddFavoriteVoteHoverIcon} alt="" />;
    }
    case 'active': {
      return <img src={AddFavoriteVoteActiveIcon} alt="" />;
    }
    default: {
      return <img src={AddFavoriteVoteDefaultIcon} alt="" />;
    }
  }
};
