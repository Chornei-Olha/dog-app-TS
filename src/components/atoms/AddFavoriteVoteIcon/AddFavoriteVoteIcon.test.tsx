import { composeStories } from '@storybook/react';
import * as stories from './AddFavoriteVoteIcon.stories';
import { testSnapshot } from '../../../test/utils/snapshot.utils';

describe('Icon', () => {
  const { Primary } = composeStories(stories);
  testSnapshot(<Primary />);
});
