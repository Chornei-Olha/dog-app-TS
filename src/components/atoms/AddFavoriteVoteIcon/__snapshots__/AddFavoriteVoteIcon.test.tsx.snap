// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddFavoriteVoteIcon should match snapshot Primary 1`] = `
<div>
  <img
    alt=""
    src="test-file-stub"
  />
</div>
`;
